process/smartContract/scQueryService.go

```diff
@@ -206,7 +206,11 @@ func (service *SCQueryService) executeScCall(query *process.SCQuery, gasPrice ui
 		if err != nil {
 			return nil, nil, err
 		}
-		service.blockChainHook.SetCurrentHeader(blockHeader)
+
```

Comment:
this won't work correctly, as the header that is set for the SC query service is can be an old header.
The trigger follows in this implementation the up to date chain data (the processing epoch start trigger is used), and this will cause the blockchain hook epoch start header to not be correctly configured.